src/megaclient.cpp

```diff
@@ -871,6 +871,49 @@ void MegaClient::acknowledgeuseralerts()
     useralerts.acknowledgeAll();
 }
 
+void MegaClient::activateoverquota(dstime timeleft)
+{
+    if (timeleft)
+    {
+        LOG_warn << "Bandwidth overquota";
+        overquotauntil = Waiter::ds + timeleft;
+        for (int d = GET; d == GET || d == PUT; d += PUT - GET)
+        {
+            for (transfer_map::iterator it = transfers[d].begin(); it != transfers[d].end(); it++)
+            {
+                Transfer *t = it->second;
+                t->bt.backoff(timeleft);
+                if (t->slot && (t->state != TRANSFERSTATE_RETRYING
+                                || !t->slot->retrying
+                                || t->slot->retrybt.nextset() != overquotauntil))
+                {
+                    t->state = TRANSFERSTATE_RETRYING;
+                    t->slot->retrybt.backoff(timeleft);
+                    t->slot->retrying = true;
+                    app->transfer_failed(t, API_EOVERQUOTA, timeleft);
+                }
+            }
+        }
+    }
+    else if (setstoragestatus(STORAGE_RED))
+    {
+        LOG_warn << "Storage overquota";
+        for (transfer_map::iterator it = transfers[PUT].begin(); it != transfers[PUT].end(); it++)
+        {
+            Transfer *t = it->second;
+            t->bt.backoff(NEVER);
+            if (t->slot)
+            {
+                t->state = TRANSFERSTATE_RETRYING;
+                t->slot->retrybt.backoff(NEVER);
+                t->slot->retrying = true;
+                app->transfer_failed(t, API_EOVERQUOTA, 0);
+            }
```

Comment:
In case of `putnodes()` relative to an upload, the API seems to not reject the provided upload-token despite being overquota. However, this PR handles that case by calling `activateoverquota(0)` already, but testing by simulate a `API_EOVERQUOTA` in response of the command is worthy, to check the transfer resumes appropriately.

Reply:
The overquota state is enabled in `CommandPutNodes::procresult` to cover those cases.
I will simulate that case to ensure that it's correctly managed 👍